exams/tasks_test.py

```diff
@@ -1,57 +1,136 @@
         """
         Verify that export_exam_profiles makes calls to export the pending profiles
         """
-        from exams.tasks import export_exam_profiles
+        assert ExamProfile.objects.count() == 10
 
-        with mute_signals(post_save):
-            exam_profiles = ExamProfileFactory.create_batch(10, status=ExamProfile.PROFILE_PENDING)
+        # use side_effect to assert at call-time because query return values
+        # mutate by the time export_exam_profiles returns
+        def side_effect(tsv, qs):  # pylint: disable=missing-docstring
+            assert hasattr(tsv, 'write')
+            assert isinstance(qs, QuerySet)
 
-        valid, invalid = exam_profiles[:5], exam_profiles[5:]
+            profiles = list(qs)
+            assert len(profiles) == 10
+            for profile in profiles:
+                assert profile in self.all_profiles
 
-        cdd_writer_mock.return_value = (valid, invalid)
+            return DEFAULT
 
-        export_exam_profiles()
+        with patch(
+            'exams.pearson.write_cdd_file',
+            return_value=(self.valid_profiles, self.invalid_profiles),
+            side_effect=side_effect,
+        ) as write_cdd_file_mock:
+            export_exam_profiles()
 
         assert upload_tsv_mock.call_count == 1
         assert 'cdd-' in upload_tsv_mock.call_args[0][0]
         assert upload_tsv_mock.call_args[0][0].endswith('.dat')
 
-        assert cdd_writer_mock.call_count == 1
-        assert hasattr(cdd_writer_mock.call_args[0][0], 'write')  # was first arg a file-like object?
-        assert isinstance(cdd_writer_mock.call_args[0][1], QuerySet)
-        assert all([isinstance(ep, ExamProfile) for ep in cdd_writer_mock.args[0][1]])
+        assert write_cdd_file_mock.call_count == 1
+
+        invalid_profiles = ExamProfile.objects.filter(status=ExamProfile.PROFILE_INVALID)
+        in_progress_profiles = ExamProfile.objects.filter(status=ExamProfile.PROFILE_IN_PROGRESS)
+
+        assert len(invalid_profiles) == 5
+        assert len(in_progress_profiles) == 5
+
+        # invalid profiles should be marked as such
+        for exam_profile in self.invalid_profiles:
+            assert exam_profile in invalid_profiles
+
+        # valid profiles should be in progress
+        for exam_profile in self.valid_profiles:
+            assert exam_profile in in_progress_profiles
+
+
+class ExamAuthorizationTasksTest(TestCase):
+    """
+    Tests for exam authorization tasks
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        with mute_signals(post_save):
+            cls.exam_auths = ExamAuthorizationFactory.create_batch(10, status=ExamAuthorization.STATUS_PENDING)
+
+    @patch('exams.pearson.upload_tsv')
+    def test_export_exam_authorizations(self, upload_tsv_mock):  # pylint: disable=no-self-use
+        """
+        Verify that export_exam_authorizations exports pending exam auths
+        """
+        assert ExamAuthorization.objects.count() == 10
```

Comment:
I think my comment is more about why exactly `10`: I think you should assert equal to the length of the elements in the setup or something else easily identifiable